src/System.Net.Http/src/System/Net/Http/HttpMethod.cs

```diff
@@ -178,5 +178,16 @@ internal bool MustHaveRequestBody
                        !ReferenceEquals(this, HttpMethod.Options) && !ReferenceEquals(this, HttpMethod.Delete);
             }
         }
+
+        internal bool NoResponseBody
+        {
+            get
+            {
+                // Normalize before calling this
+                Debug.Assert(ReferenceEquals(this, Normalize(this)));
+
+                return ReferenceEquals(this, HttpMethod.Head) || ReferenceEquals(this, HttpMethod.Put);
```

Comment:
I don't believe PUT is valid here.  Whereas the HTTP spec specifies that a "server MUST NOT return a message-body in the response", it makes no such claim for PUT.  For reference, with regards to response body, SocketsHttpHandler special-cases HEAD but not PUT:
https://github.com/dotnet/corefx/blob/a0dbd526003f5ac19761bdd5a25dd8476009bdb0/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs#L653

Reply:
Okay, I took out the PUT.

Reply:
This came up in discussion.  It would seem to make sense to avoid allocating the buffer if the response status code was 204 (No Content) which would likely solve the well formed PUT case as well

Reply:
@lewing, in what scenario would you get back a 204 with a Content-Length?